app/components/primer/octicon_symbol_component.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require "octicons"
+
+module Primer
+  # OcticonSymbol renders a symbol dictionary using a list of <%= link_to_octicons %>.
+  class OcticonSymbolComponent < Primer::Component
+    # Required list of icons. These will be the icons that are rendered.
+    #
+    # @param symbol [String] Name of <%= link_to_octicons %> to use.
+    # @param size [Symbol] <%= one_of(Primer::OcticonComponent::SIZE_MAPPINGS) %>
+    renders_many :icons, "Symbol"
+
+    # @example Symbol dictionary
+    #   <%= render(Primer::OcticonComponent.new(icon: :check, use_symbol: true, color: :icon_success)) %>
+    #   <%= render(Primer::OcticonComponent.new(icon: :check, use_symbol: true, color: :text_danger)) %>
+    #   <%= render(Primer::OcticonComponent.new(icon: :check, use_symbol: true, size: :medium)) %>
+    #   <%= render(Primer::OcticonSymbolComponent.new) do |c| %>
+    #     <%= c.icon(symbol: :check) %>
+    #     <%= c.icon(symbol: :check, size: :medium) %>
+    #   <% end %>
+    def initialize
+      super
+    end
+
+    def before_render
+      # Make sure we don't have any duplicate icons
+      icons.uniq! { |icon| "#{icon.icon.symbol}#{icon.icon.height}" }
```

Comment:
There's a possibility that someone might try and get a size of an icon that doesn't exist. In that case the library will default to the closest size. Which could end up with the same symbol in the defs. This won't break anything with rendering, but seemed like a waste of resources?